xdem/spstats.py

```diff
@@ -885,22 +886,29 @@ def patches_method(dh : np.ndarray, mask: np.ndarray[bool], gsd : float, area_si
     return df
 
 
-def plot_vgm(df: pd.DataFrame, fit_fun: Callable = None):
+def plot_vgm(df: pd.DataFrame, list_fit_fun: Optional[list[Callable]] = None, list_fit_fun_label: Optional[list[str]] = None):
 
     fig, ax = plt.subplots(1)
```

Comment:
Completely up to you if you want to include it, but I've added an `ax: plt.Axes = plt.gca()` argument to many of my personal plotting projects, this allows multiple types of calls, as it either plots on a given axis, or defaults to the current one:

```python
plt.subplot(131)
plot_vgm(...)  # This will plot on subplot 121
ax2 = plt.subplot(132)
plot_vgm(...)  # This will plot on subplot 122
ax3 = plt.subplot(133)
plot_vgm(..., ax=ax2)  # This will also plot on 122. Madness!

```

Reply:
Yes, quite familiar with massive plotting exercises... é_è

Correct me if I'm wrong: doing this would mean that we'd have to always initatilizing `ax` outside the function when we don't want to plot on an existing one, though?
Or could we add a loop at the start of each function to check if `ax` exists, and create it if it doesn't?

Reply:
`plt.gca()` means "Get the Current Axis (gca), or make one if there is not current axis", so there's no need for any other check!

Reply:
You can do the same way as I do it in georaster: https://github.com/GlacioHack/GeoUtils/blob/12bd607fe8024720ff264545abf0cc5ba0156ea6/geoutils/georaster.py#L1242
So, if none specified, a new figure is created, otherwise uses the ax that is passed.

Reply:
Added